drift/info_parser.py

```diff
@@ -140,56 +155,27 @@ def _group_comparisons(comparisons):
 
     # default values for method run
     grouped_comparisons = []
+    grouped_comparisons_with_dotnotation = {}
     processed_group_names = set()
 
-    def _get_group_name(name):
-        n, _, _ = name.partition(".")
-        return n
-
-    def _get_value_name(name):
-        _, _, n = name.partition(".")
-        return n
-
-    def _find_group(name, grouped_comps):
-        for group in grouped_comps:
-            if group["name"] == name:
-                return group
-
-    def _set_summary_state(grouped_comparison):
-        states = {comparison["state"] for comparison in grouped_comparison["comparisons"]}
-        if COMPARISON_DIFFERENT in states:
-            grouped_comparison["state"] = COMPARISON_DIFFERENT
-        elif COMPARISON_INCOMPLETE_DATA in states:
-            if COMPARISON_SAME in states:
-                grouped_comparison["state"] = COMPARISON_DIFFERENT
-            else:
-                grouped_comparison["state"] = COMPARISON_INCOMPLETE_DATA
-        elif COMPARISON_SAME in states and len(states) == 1:
-            grouped_comparison["state"] = COMPARISON_SAME
-        else:  # use 'incomplete data' as the fallback state if something goes wrong
-            grouped_comparison["state"] = COMPARISON_INCOMPLETE_DATA
-
-    # with PT_GC_BUILD_GROUP_NAMES.time(): # move to proper place or remove competely
-    # with PT_GC_POPULATE_GROUPS.time(): # move to proper place or remove competely
-    # with PT_GC_SET_SUMMARY.time(): # move to proper place or remove competely
     for comparison in comparisons:
         if "." in comparison["name"]:
-            group_name = _get_group_name(comparison["name"])
+            group_name, _, comparison_value_name = comparison["name"].partition(".")
+
             built_group = {"name": group_name, "comparisons": []}
 
             # build out group names
             if group_name not in processed_group_names:
                 processed_group_names.add(group_name)
-                grouped_comparisons.append(built_group)
+                grouped_comparisons_with_dotnotation.update({group_name: built_group})
```

Comment:
Can this be done without calling update?

Reply:
I'm afraid not, since we're adding a new `dict`, the `update` is the method to include it.

Reply:
How about direct assignment by key?
```
grouped_comparisons_with_dotnotation[group_name] = ...
```

Reply:
I tested here, and to be honest doesn't take much affect into the time.

And, using direct assignment have a side affect also (look at the screenshots).. **only in that case**, we don't need the `dict` with a key.. we need a dict without a key to build the comparisons. We can move with the direct assignment, but to do that, we need to change all the logic in the code, I don't know if the result compensates for all this change. 

Using update:

![Screen Shot 2022-08-10 at 10 38 55](https://user-images.githubusercontent.com/2904206/183915681-7b59dafc-4e8a-43e8-b7d6-ceee05c931a3.png)

Using direct assignment:
![Screen Shot 2022-08-10 at 10 35 07](https://user-images.githubusercontent.com/2904206/183915555-ff5104d6-7fbb-410f-87b4-9b455a651102.png)

Reply:
Sorry I do not understand the reasoning. Direct assignment and update would have the same result. The only difference is that for the update a new object has to be created that is iterable, then iterated on and then added to dictionary. I acknowledge that the save on speed would not be much, however with direct assignment it reads better and uses one less objects that needs to be GCd.

```
>>> d = {}
>>> built_group = {"name": "g1", "comparisons": []}
>>> d.update({"g1": built_group})
>>> built_group = {"name": "g2", "comparisons": []}
>>> d["g2"] = built_group
>>> d
{'g1': {'name': 'g1', 'comparisons': []}, 'g2': {'name': 'g2', 'comparisons': []}}
```